cmd/cluster-display/main.go

```diff
@@ -118,47 +118,66 @@ func getClusterPoolPage(ctx context.Context, hiveClient ctrlruntimeclient.Client
 	return &page, nil
 }
 
-func getClusterPage(ctx context.Context, clients map[string]ctrlruntimeclient.Client, skipHive bool) (*Page, error) {
+type ClusterInfoGetter interface {
+	GetClusterDetails(ctx context.Context, cluster string, client ctrlruntimeclient.Client) (map[string]string, error)
+}
+
+type clusterInfoGetter struct{}
```

Comment:
I think since you are going to have this struct, you should add a `cluster` field to it. That way you don't also have to pass the cluster to the method. You would then just create one of these for each cluster in `getClusterPage`